### PR TITLE
Pulling in grid.js and css files from nightly.

### PR DESCRIPTION
**Pull request by emtwo**

### CODE REVIEW

File: css/newTab.css [hunk 1]
```diff
@@ -14,6 +14,7 @@ body {
   padding: 0;
   margin: 0;
   background-color: #F9F9F9;
+  display: -moz-box;
```
Comment (marcoscaceres) [inline css/newTab.css:17]: Just out of curiosity, what does this do?

File: css/newTab.css [hunk 2]
```diff
@@ -522,7 +526,7 @@ input[type=button] {
   border-radius: 5px 5px 0px 0px;
   max-width: 300px;
   overflow: hidden;
-  display: block;
+  display: table-cell;
```
Comment (marcoscaceres) [inline css/newTab.css:529]: hmmm... table-cell? Why doesn't something like inline-block work?

File: js/grid.js [hunk 3]
```diff
@@ -234,15 +245,50 @@
...
-      let clientHeight = document.documentElement.clientHeight;
-      let availSpace = clientHeight - this._cellMargin -
-        document.querySelector("#newtab-search-container").offsetHeight;
-      let visibleRows = Math.floor(availSpace / this._cellHeight);
-      let columns = Math.max(1, gNewTab.columns);
+
+      let searchContainer = document.querySelector("#newtab-search-container");
+      // Save search-container margin height
+      if (this._searchContainerMargin  === undefined) {
+        this._searchContainerMargin = parseFloat(getComputedStyle(searchContainer).marginBottom) +
```
Comment (marcoscaceres) [inline js/grid.js:252]: Nit, would be a bit easier to read:

``` JS
var searchStyles = getComputedStyle(searchContainer);
this._searchContainerMargin = parseFloat(seachStyles.marginBottom) +  parseFloat(seachStyles.marginTop); 
```

File: js/grid.js [hunk 4]
```diff
@@ -215,6 +216,16 @@
...
+     */
+    _isHistoricalTile(aPos) {
+      let site = this.sites[aPos];
+      return site && (site.isPinned() || site.link && site.link.type === "history");
+    },
+
+    /**
      * Make sure the correct number of rows and columns are visible
      */
     _resizeGrid() {
```
Comment (marcoscaceres) [inline js/grid.js:231]: This all seems like it's just replicating the behavior that should be handled by flexbox? We should see if we can get rid of this using CSS.

Comment (emtwo) [inline js/grid.js:231]: Well, `resizeGrid` is taking into account the user-set values for grid rows and columns, it's only adding a scrollbar for users who have extra rows that contain history or pinned tiles, etc. I think it's doing more than what a flexbox can do...

Comment (marcoscaceres) [inline js/grid.js:231]: oki, thanks for the clarification.

**Comment by marcoscaceres** [comment]
R+

**Pull request merged**